src/gt4py/next/program_processors/runners/dace_fieldview/gtir_builtin_symbol_ref.py

```diff
@@ -0,0 +1,86 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2023, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+
+from typing import Optional
+
+import dace
+
+from gt4py.next.iterator import ir as itir
+from gt4py.next.program_processors.runners.dace_fieldview.gtir_dataflow_builder import (
+    GtirDataflowBuilder,
+)
+from gt4py.next.program_processors.runners.dace_fieldview.utility import as_dace_type
+from gt4py.next.type_system import type_specifications as ts
+
+
+class GtirBuiltinSymbolRef(GtirDataflowBuilder):
+    """Generates the dataflow subgraph for a `itir.SymRef` node."""
+
+    _sym_name: str
+    _sym_type: ts.FieldType | ts.ScalarType
+
+    def __init__(
+        self,
+        sdfg: dace.SDFG,
+        state: dace.SDFGState,
+        data_types: dict[str, ts.FieldType | ts.ScalarType],
+        node: itir.SymRef,
+    ):
+        super().__init__(sdfg, state, data_types)
+        sym_name = str(node.id)
+        assert sym_name in self._data_types
+        self._sym_name = sym_name
+        self._sym_type = self._data_types[sym_name]
+
+    def _get_access_node(self) -> Optional[dace.nodes.AccessNode]:
+        """Returns, if present, the access node in current state for the data symbol."""
+        access_nodes = [
+            node
+            for node in self._state.nodes()
+            if isinstance(node, dace.nodes.AccessNode) and node.data == self._sym_name
+        ]
+        if len(access_nodes) == 0:
+            return None
+        assert len(access_nodes) == 1
+        return access_nodes[0]
+
+    def _build(self) -> list[tuple[dace.nodes.Node, ts.FieldType | ts.ScalarType]]:
+        sym_node = self._get_access_node()
+        if sym_node:
+            # if already present in current state, share access node
+            pass
+
+        elif isinstance(self._sym_type, ts.FieldType):
```

Comment:
Is this correct? If the call to `_get_access_node()` returns something a node instead of `None`, I would assume that it would go through the `pass` branch so the `isinstance()` check would never be `True`. Am I reading this wrong?

Reply:
Correct. If the access node is already present in current state, we can simply return it at the end of this function.